src/validator_tests.zig

```diff
@@ -0,0 +1,245 @@
+    try expect(validator.slashed == deserialized.slashed);
+    try expect(validator.activation_eligibility_epoch == deserialized.activation_eligibility_epoch);
+    try expect(validator.activation_epoch == deserialized.activation_epoch);
+    try expect(validator.exit_epoch == deserialized.exit_epoch);
+    try expect(validator.withdrawable_epoch == deserialized.withdrawable_epoch);
+}
+
+test "Validator struct hash tree root" {
+    const validator = Validator{
+        .pubkey = [_]u8{0x01} ** 48,
+        .withdrawal_credentials = [_]u8{0x02} ** 32,
+        .effective_balance = 32000000000,
+        .slashed = true,
+        .activation_eligibility_epoch = 1,
+        .activation_epoch = 2,
+        .exit_epoch = 100,
+        .withdrawable_epoch = 200,
+    };
+
+    var hash: [32]u8 = undefined;
+    try hashTreeRoot(Validator, validator, &hash, std.testing.allocator);
+
+    // Hash should be deterministic for the same validator
+    var hash2: [32]u8 = undefined;
+    try hashTreeRoot(Validator, validator, &hash2, std.testing.allocator);
+    try expect(std.mem.eql(u8, &hash, &hash2));
+
+    // Different validator should produce different hash
+    const validator2 = Validator{
+        .pubkey = [_]u8{0xFF} ** 48,
+        .withdrawal_credentials = [_]u8{0x02} ** 32,
+        .effective_balance = 32000000000,
+        .slashed = true,
+        .activation_eligibility_epoch = 1,
+        .activation_epoch = 2,
+        .exit_epoch = 100,
+        .withdrawable_epoch = 200,
+    };
+
+    var hash3: [32]u8 = undefined;
+    try hashTreeRoot(Validator, validator2, &hash3, std.testing.allocator);
+    try expect(!std.mem.eql(u8, &hash, &hash3));
+}
+
+test "List[Validator] serialization and hash tree root" {
+    const MAX_VALIDATORS = 100;
+    const ValidatorList = utils.List(Validator, MAX_VALIDATORS);
+
+    var validator_list = try ValidatorList.init(0);
+
+    // Add test validators
+    const validator1 = Validator{
+        .pubkey = [_]u8{0x01} ** 48,
+        .withdrawal_credentials = [_]u8{0x11} ** 32,
+        .effective_balance = 32000000000,
+        .slashed = false,
+        .activation_eligibility_epoch = 0,
+        .activation_epoch = 0,
+        .exit_epoch = 18446744073709551615,
+        .withdrawable_epoch = 18446744073709551615,
+    };
+
+    const validator2 = Validator{
+        .pubkey = [_]u8{0x02} ** 48,
+        .withdrawal_credentials = [_]u8{0x22} ** 32,
+        .effective_balance = 31000000000,
+        .slashed = false,
+        .activation_eligibility_epoch = 1,
+        .activation_epoch = 1,
+        .exit_epoch = 18446744073709551615,
+        .withdrawable_epoch = 18446744073709551615,
+    };
+
+    try validator_list.append(validator1);
+    try validator_list.append(validator2);
+
+    // Test serialization
+    var list = ArrayList(u8).init(std.testing.allocator);
+    defer list.deinit();
+    try serialize(ValidatorList, validator_list, &list);
```

Comment:
we need a bytes match against bytes generated from another tool (so yes hardcoded bytes)